Source/Core/Core/Boot/Boot.cpp

```diff
@@ -271,199 +312,187 @@ bool CBoot::Load_BS2(const std::string& boot_rom_filename)
+        {
+          NOTICE_LOG(BOOT, "Loading default ISO %s", config.m_strDefaultISO.c_str());
+          volume = SetDisc(DiscIO::CreateVolumeFromFilename(config.m_strDefaultISO));
+        }
+
+        // Poor man's bootup
+        if (config.bWii)
+        {
+          HID4.SBE = 1;
+          SetupMSR();
+          SetupBAT(config.bWii);
+
+          // Because there is no TMD to get the requested system (IOS) version from,
+          // we default to IOS58, which is the version used by the Homebrew Channel.
+          SetupWiiMemory(volume, 0x000000010000003a);
+        }
+        else
+        {
+          EmulatedBS2_GC(volume, true);
+        }
+
+        Load_FST(config.bWii, volume);
+        dolLoader.Load();
+        PC = dolLoader.GetEntryPoint();
+
+        if (LoadMapFromFilename())
+          HLE::PatchFunctions();
+      }
+
+      if (executable.type == BootParameters::Executable::Type::ELF)
+      {
+        const DiscIO::IVolume* volume = SetDefaultDisc();
+
+        // Poor man's bootup
+        if (config.bWii)
+        {
+          // Because there is no TMD to get the requested system (IOS) version from,
+          // we default to IOS58, which is the version used by the Homebrew Channel.
+          SetupWiiMemory(volume, 0x000000010000003a);
+        }
+        else
+        {
+          EmulatedBS2_GC(volume, true);
+        }
+
+        Load_FST(config.bWii, volume);
+        if (!Boot_ELF(executable.executable_path))
+          return false;
+
+        // Note: Boot_ELF calls HLE::PatchFunctions()
+
+        UpdateDebugger_MapLoaded();
+        Dolphin_Debugger::AddAutoBreakpoints();
+      }
 
-      // Because there is no TMD to get the requested system (IOS) version from,
-      // we default to IOS58, which is the version used by the Homebrew Channel.
-      SetupWiiMemory(volume, 0x000000010000003a);
+      return true;
     }
-    else
+
+    bool operator()(const BootParameters::NAND& nand) const
     {
-      EmulatedBS2_GC(volume, true);
+      NOTICE_LOG(BOOT, "Booting from NAND: %s", nand.content_path.c_str());
+      SetDefaultDisc();
+      return Boot_WiiWAD(nand.content_path);
     }
 
-    Load_FST(dolWii, volume);
-    dolLoader.Load();
-    PC = dolLoader.GetEntryPoint();
+    bool operator()(const BootParameters::IPL& ipl) const
+    {
+      NOTICE_LOG(BOOT, "Booting GC IPL: %s", ipl.ipl_path.c_str());
+      if (!File::Exists(ipl.ipl_path))
+      {
+        if (ipl.disc)
+          PanicAlertT("Cannot start the game, because the GC IPL could not be found.");
+        else
+          PanicAlertT("Cannot find the GC IPL.");
+        return false;
+      }
 
-    if (LoadMapFromFilename())
-      HLE::PatchFunctions();
+      if (!Load_BS2(ipl.ipl_path))
+        return false;
 
-    break;
-  }
+      if (ipl.disc)
+      {
+        NOTICE_LOG(BOOT, "Inserting disc: %s", ipl.disc->image_path.c_str());
+        SetDisc(DiscIO::CreateVolumeFromFilename(ipl.disc->image_path));
+      }
 
```

Comment:
On IRC, it was decided to just leave this as is, as this is already the current behaviour.